File: bench/ports/kotlin/objects.kt
data class Point(val x: Long, val y: Long)

fun manhattan(p: Point): Long = Math.abs(p.x) + Math.abs(p.y)

fun main() {
    var sum = 0L
    for (i in 0 until 10000000L) {
        val p = Point(i % 100 - 50, i % 37 - 18)
        sum += manhattan(p)
    }
    println(sum)
}

File: bench/ports/rust/objects.rs
struct Point { x: i64, y: i64 }
fn manhattan(p: &Point) -> i64 { p.x.abs() + p.y.abs() }
fn main() {
    let mut sum: i64 = 0;
    for i in 0..10_000_000i64 {
        let p = Point { x: i % 100 - 50, y: i % 37 - 18 };
        sum += manhattan(&p);
    }
    println!("{}", sum);
}
